tests/server/core/hot_region_storage_test.go

```diff
@@ -143,3 +143,167 @@ func (s *hotRegionHistorySuite) TestHotRegionStorage(c *C) {
+		{
+			Id:            2,
+			State:         metapb.StoreState_Up,
+			LastHeartbeat: time.Now().UnixNano(),
+		},
+	}
+
+	leaderServer := cluster.GetServer(cluster.GetLeader())
+	c.Assert(leaderServer.BootstrapCluster(), IsNil)
+	for _, store := range stores {
+		pdctl.MustPutStore(c, leaderServer.GetServer(), store)
+	}
+	defer cluster.Destroy()
+	startTime := time.Now().UnixNano() / int64(time.Millisecond)
+	pdctl.MustPutRegion(c, cluster, 1, 1, []byte("a"), []byte("b"), core.SetWrittenBytes(3000000000), core.SetReportInterval(statistics.WriteReportInterval))
+	// wait hot scheduler starts
+	time.Sleep(5000 * time.Millisecond)
+	endTime := time.Now().UnixNano() / int64(time.Millisecond)
+	storage := leaderServer.GetServer().GetHistoryHotRegionStorage()
+	iter := storage.NewIterator([]string{core.WriteType.String()}, startTime, endTime)
+	next, err := iter.Next()
+	c.Assert(next, NotNil)
+	c.Assert(err, IsNil)
+	c.Assert(next.RegionID, Equals, uint64(1))
+	c.Assert(next.StoreID, Equals, uint64(1))
+	c.Assert(next.HotRegionType, Equals, core.WriteType.String())
+	next, err = iter.Next()
+	c.Assert(err, IsNil)
+	c.Assert(next, IsNil)
+	schedule := leaderServer.GetConfig().Schedule
+	// set reserved day to zero,close hot region storage
+	schedule.HotRegionsReservedDays = 0
+	leaderServer.GetServer().SetScheduleConfig(schedule)
+	time.Sleep(3 * interval)
+	pdctl.MustPutRegion(c, cluster, 2, 2, []byte("c"), []byte("d"), core.SetWrittenBytes(6000000000), core.SetReportInterval(statistics.WriteReportInterval))
+	time.Sleep(10 * interval)
+	endTime = time.Now().UnixNano() / int64(time.Millisecond)
+	storage = leaderServer.GetServer().GetHistoryHotRegionStorage()
+	iter = storage.NewIterator([]string{core.WriteType.String()}, startTime, endTime)
+	next, err = iter.Next()
+	c.Assert(next, NotNil)
+	c.Assert(err, IsNil)
+	c.Assert(next.RegionID, Equals, uint64(1))
+	c.Assert(next.StoreID, Equals, uint64(1))
+	c.Assert(next.HotRegionType, Equals, core.WriteType.String())
+	next, err = iter.Next()
+	c.Assert(err, IsNil)
+	c.Assert(next, IsNil)
+	// set reserved day to one,open hot region storage
+	schedule.HotRegionsReservedDays = 1
+	leaderServer.GetServer().SetScheduleConfig(schedule)
+	time.Sleep(3 * interval)
+	endTime = time.Now().UnixNano() / int64(time.Millisecond)
+	storage = leaderServer.GetServer().GetHistoryHotRegionStorage()
+	iter = storage.NewIterator([]string{core.WriteType.String()}, startTime, endTime)
+	next, err = iter.Next()
+	c.Assert(next, NotNil)
+	c.Assert(err, IsNil)
+	c.Assert(next.RegionID, Equals, uint64(1))
+	c.Assert(next.StoreID, Equals, uint64(1))
+	c.Assert(next.HotRegionType, Equals, core.WriteType.String())
+	next, err = iter.Next()
+	c.Assert(next, NotNil)
+	c.Assert(err, IsNil)
+	c.Assert(next.RegionID, Equals, uint64(2))
+	c.Assert(next.StoreID, Equals, uint64(2))
+	c.Assert(next.HotRegionType, Equals, core.WriteType.String())
+}
+
+func (s *hotRegionHistorySuite) TestHotRegionStorageWriteIntervalConfigChange(c *C) {
```

Comment:
the unit test case is similar with TestHotRegionStorageWriteIntervalConfigChange, it will be better if  exract the some code.